### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by ualaska-se**

### CODE REVIEW

File: libraries/python_install_helper.rb [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+    end
+
+    def generate_configure_options(install_directory)
+      code = ''
+      code += ' --enable-shared'
+      code += " --prefix=#{install_directory}"
+      code += " --exec_prefix=#{install_directory}"
+      code += ' --with-system-ffi'
+      # Optimizations are broken on ancient CentOS package versions
+      code += ' --enable-optimizations' # if node['platform_family'] == 'debian'
```
Comment (Arsh25) [inline libraries/python_install_helper.rb]: Was the if meant to be commented out? If yes, then the broken on centOS comment makes no sense

File: libraries/python_install_helper.rb [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+      return File.join(new_resource.build_directory, base) if new_resource.build_directory
+
+      create_default_directories
+      return File.join('/var/chef/cache', base)
+    end
+
+    def clear_source_directory(build_directory, new_resource)
+      dir = build_directory
+      bash 'Clear Archive' do
+        code "rm -rf #{dir}\nmkdir #{dir}\nchmod #{new_resource.owner} #{dir}\nchgrp #{new_resource.group} #{dir}"
```
Comment (Arsh25) [inline libraries/python_install_helper.rb]: Any reason you don't just do a `chown #{new_resource.owner}: #{new_resource.group} #{dir}`

File: libraries/python_install_helper.rb [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+
+      return 'tar xzf' if filename.match?(/\.(:?tar\.gz|tgz)/)
+
+      raise "Archive not supported: #{filename}"
+    end
+
+    def code_for_extraction(download_file, build_directory, new_resource)
+      code = <<~CODE
+        #{extract_command(download_file)} #{download_file}
+        chown -R #{new_resource.owner} #{build_directory}
```
Comment (Arsh25) [inline libraries/python_install_helper.rb]: could just do `chown -R #{new_resource.owner}:#{new_resource.group} #{build_directory}`

File: libraries/python_install_helper.rb (hunk 2 above)
Comment (ualaska-se) [reply libraries/python_install_helper.rb]: Because I did not know that command!

File: test/integration/default/default_test.rb [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+LOCAL_GROUP =
+  if node['platform'] == 'debian'
+    'staff'
+  else
+    'root'
+  end
+
+LOCAL_MODE =
+  if node['platform'] == 'debian'
+    0o2755
```
Comment (Arsh25) [inline test/integration/default/default_test.rb:14]: wait...What does this mean?

File: test/integration/default/default_test.rb [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+describe bash "/usr/local/#{BASE_NAME}/bin/#{BASE_NAME}3 --version" do
+  its(:exit_status) { should eq 0 }
+  its(:stderr) { should eq '' }
+  its(:stdout) { should match(/3\.6\.9/) }
+end
+
+describe bash "/opt/#{BASE_NAME}/#{CURR_VER}/bin/#{BASE_NAME}3 -c 'import ssl; print(ssl.OPENSSL_VERSION)'" do
+  its(:exit_status) { should eq 0 }
+  its(:stderr) { should eq '' }
+  its(:stdout) { should_not match(/1\.1\.1c/) } if node['platform_family'] == 'debian'
```
Comment (Arsh25) [inline test/integration/default/default_test.rb:501]: Version check on Debian?

File: test/integration/default/default_test.rb (hunk 4 above)
Comment (ualaska-se) [reply test/integration/default/default_test.rb:14]: Debian sets the sticky bit so all files are grouped into staff

**Review by Arsh25 — APPROVED** [review]

File: test/integration/default/default_test.rb (hunk 5 above)
Comment (ualaska-se) [reply test/integration/default/default_test.rb:501]: Debian uses the system OpenSSL; CentOS libs were ancient

File: libraries/python_install_helper.rb (hunk 1 above)
Comment (ualaska-se) [reply libraries/python_install_helper.rb]: I eliminated the obsolete code

**Pull request merged**